### PR TITLE
Add build-in function for botbuilder-python

### PR DESCRIPTION
**Pull request by ShYuPe**
Add build-in function for botbuilder-python
Still working on it...

### CODE REVIEW

File: libraries/adaptive-expressions/adaptive/expressions/expression_functions.py [hunk 1]
```diff
@@ -14,6 +16,8 @@ def get_standard_functions() -> dict:
     functions.append(Subtract())
     functions.append(Multiply())
     functions.append(Divide())
+    functions.append(Equal())
+    functions.append(LessThen())
```
Comment (feich-ms) [inline libraries/adaptive-expressions/adaptive/expressions/expression_functions.py:20]: Maybe we can add these to new lines

File: libraries/adaptive-expressions/adaptive/expressions/function_utils.py [hunk 2]
```diff
@@ -234,3 +247,17 @@ def set_property(instance: object, property: str, val: object):
...
+
+    @staticmethod
+    def try_parse_list(value: object, out_list: list):
+        islist = False
+        out_list.clear()
+        if isinstance(value, list):
+            for i in list(value):
+                out_list.append(i)
+            islist = True
+        return islist
```
Comment (feich-ms) [inline libraries/adaptive-expressions/adaptive/expressions/function_utils.py:259]: this function is actually not necessary. just use isinstance(value, list) to check if it is list. This function is only useful in C# library

File: libraries/adaptive-expressions/adaptive/expressions/expression_functions.py (hunk 1 above)
Comment (feich-ms) [reply libraries/adaptive-expressions/adaptive/expressions/expression_functions.py:20]: like 
// Math
functions.append(Add())
functions.append(Subtract())
functions.append(Multiply())
functions.append(Divide())

// Comparisons
functions.append(Equal())
functions.append(LessThen())

---
In reply to: [459836534](https://github.com/microsoft/botbuilder-python/pull/1275#discussion_r459836534) [](ancestors = 459836534)

File: libraries/adaptive-expressions/adaptive/expressions/expression_functions.py [hunk 3]
```diff
@@ -1,8 +1,10 @@
-from .expression_type import ADD, SUBTRACT, MULTIPLY, DIVIDE
+from .expression_type import ADD, SUBTRACT, MULTIPLY, DIVIDE, EQUAL, LESSTHEN
 from .builtin_functions.add import Add
 from .builtin_functions.subtract import Subtract
 from .builtin_functions.multiply import Multiply
 from .builtin_functions.divide import Divide
+from .builtin_functions.equal import Equal
+from .builtin_functions.less_then import LessThen
```
Comment (feich-ms) [inline libraries/adaptive-expressions/adaptive/expressions/expression_functions.py:7]: add after from .builtin_functions.power import Power

File: libraries/adaptive-expressions/adaptive/expressions/expression_functions.py [hunk 4]
```diff
@@ -23,6 +27,8 @@ def get_standard_functions() -> dict:
         lookup[function.expr_type] = function
 
     lookup["add"] = lookup[ADD]
+    lookup["equal"] = lookup[EQUAL]
+    lookup["lessthen"] = lookup[LESSTHEN]
```
Comment (feich-ms) [inline libraries/adaptive-expressions/adaptive/expressions/expression_functions.py:31]: also here to new lines like C#

File: libraries/adaptive-expressions/adaptive/expressions/expression_functions.py (hunk 1 above)
Comment (ShYuPe) [reply libraries/adaptive-expressions/adaptive/expressions/expression_functions.py]: fixed

File: libraries/adaptive-expressions/adaptive/expressions/expression_functions.py (hunk 3 above)
Comment (ShYuPe) [reply libraries/adaptive-expressions/adaptive/expressions/expression_functions.py]: fixed

File: libraries/adaptive-expressions/adaptive/expressions/expression_functions.py (hunk 4 above)
Comment (ShYuPe) [reply libraries/adaptive-expressions/adaptive/expressions/expression_functions.py]: fixed

File: libraries/adaptive-expressions/adaptive/expressions/function_utils.py (hunk 2 above)
Comment (ShYuPe) [reply libraries/adaptive-expressions/adaptive/expressions/function_utils.py]: fixed

File: libraries/adaptive-expressions/adaptive/expressions/expression_functions.py [hunk 5]
```diff
@@ -1,15 +1,27 @@
-from .expression_type import ADD, SUBTRACT, MULTIPLY, DIVIDE
+from .expression_type import ADD, SUBTRACT, MULTIPLY, DIVIDE, EQUAL, LESSTHAN, LESSTHANOREQUAL, GREATERTHAN, GREATERTHANOREQUAL, NOT, NOTEQUAL, EXIST, OR, AND
```
Comment (feich-ms) [inline libraries/adaptive-expressions/adaptive/expressions/expression_functions.py:1]: >GREATERTHAN [](start = 96, length = 11)

remove

Comment (ShYuPe) [reply libraries/adaptive-expressions/adaptive/expressions/expression_functions.py]: fixed

**Pull request merged**